### PR TITLE
Delete range in tables

### PR DESCRIPTION
**Pull request by cammarosano**

### CODE REVIEW

File: addons/html_editor/static/src/editor/table/table_plugin.js [hunk 1]
```diff
@@ -284,15 +283,110 @@ export class TablePlugin extends Plugin {
...
+        if (firstTable.contains(startContainer)) {
+            [startContainer, startOffset] = leftPos(firstTable);
+        }
+        const lastTable = fullySelectedTables.at(-1);
+        if (lastTable.contains(endContainer)) {
+            [endContainer, endOffset] = rightPos(lastTable);
+        }
+        range = { startContainer, startOffset, endContainer, endOffset };
+
+        let { cursorPos } = this.shared.deleteRange(range);
```
Comment (cammarosano) [inline addons/html_editor/static/src/editor/table/table_plugin.js:358]: This is wrong for nested tables

File: addons/html_editor/static/src/editor/table/table_plugin.js [hunk 2]
```diff
@@ -284,15 +283,110 @@ export class TablePlugin extends Plugin {
...
+        if (areFullColumnsSelected) {
+            for (let index = firstCellColumnIndex; index <= lastCellColumnIndex; index++) {
+                this.removeColumn({ cell: firstRowCells[index] });
+            }
+            return;
+        }
+
+        if (areFullRowsSelected) {
+            for (let index = firstCellRowIndex; index <= lastCellRowIndex; index++) {
+                this.removeRow({ row: rows[index] });
```
Comment (cammarosano) [inline addons/html_editor/static/src/editor/table/table_plugin.js:328]: Remove row has an error: `this.shared.setCursorStart({ anchorNode: siblingRow, anchorOffset: 0 })` (it should take a node, not an Object as parameter)

File: addons/html_editor/static/src/editor/core/delete_plugin.js [hunk 3]
```diff
@@ -473,14 +479,25 @@ export class DeletePlugin extends Plugin {
     // Fill empty blocks
     // Remove empty inline elements, unless cursor is inside it
     handleEmptyElements(commonAncestor, cursorPos) {
+        const handleShrunkBlock = (block) => {
+            if (!isShrunkBlock(block)) {
```
Comment (ged-odoo) [inline addons/html_editor/static/src/editor/core/delete_plugin.js]: no big deal, but in this case, i would just do this check at the two locations where it's called

File: addons/html_editor/static/src/editor/table/table_plugin.js (hunk 1 above)
Comment (cammarosano) [reply addons/html_editor/static/src/editor/table/table_plugin.js:358]: I could not reproduce it, but I remember the table structure got broken after a few deleteBackwards. I have a test.todo for that.

File: addons/html_editor/static/src/editor/core/delete_plugin.js (hunk 3 above)
Comment (cammarosano) [reply addons/html_editor/static/src/editor/core/delete_plugin.js]: Ok, I've changed it.

**Pull request merged**